mesonbuild/build.py

```diff
@@ -1481,7 +1481,7 @@ def check_can_link_together(self, t: BuildTargetTypes) -> None:
         if not self.uses_rust() and links_with_rust_abi:
             raise InvalidArguments(f'Try to link Rust ABI library {t.name!r} with a non-Rust target {self.name!r}')
         if self.for_machine is not t.for_machine and (not links_with_rust_abi or t.rust_crate_type != 'proc-macro'):
-            msg = f'Tried to tied to mix a {t.for_machine} library ("{t.name}") with a {self.for_machine} target "{self.name}"'
+            msg = f'Tried to link {self.name!r} for {self.for_machine} with {t.name!r} for {t.for_machine}.'
```

Comment:
the change from explicitly quoting to using `!r` is a logically separate change, it should either be in a separate commit or dropped